scripts/lis_hyperv_platform/lisa_run.ps1

```diff
@@ -42,12 +42,34 @@ function Edit-XmlTest {
     popd
 }
 
+function Check-LisaExit {
+    param(
+        [int] $exitCode
+    )
+
+    $icaPath=".\TestResults\**\ica.log"
+    if ($exitCode) {
+        if (Test-Path $icaPath){
+            $summaryLine = Select-String -Path $icaPath -Pattern "Test Results Summary"
+            if (!$summaryLine) {
+                throw "Lisa failed"
+            } else {
+                Write-Host "Lisa finished successfully and one of the tests failed"
+                $LASTEXITCODE = 0
+            }
+        } else {
+            throw "Cannot find test log"
+        }
+    }
+}
+
 function Main {
     pushd "$WorkDir"
     ($KeyName, $XmlName) = Get-Dependencies $KeyPath $XmlTest
     Edit-XmlTest $VMName $XmlName $KeyName 
     pushd ".\lis-test\WS2012R2\lisa\"
     .\lisa.ps1 run xml\$XmlName -dbg 3
```

Comment:
isn't this code failing with an error -- don t you need to put a try/catch here?